docs/contributor/image_mode.md

```diff
@@ -0,0 +1,308 @@
+
+```
+$ find /lib/modules/$(uname -r) -name "openvswitch*"
+/lib/modules/6.9.9-200.fc40.x86_64/kernel/net/openvswitch
+/lib/modules/6.9.9-200.fc40.x86_64/kernel/net/openvswitch/openvswitch.ko.xz
+
+$ IMAGE_NAME=microshift-4.16-bootc
+$ sudo podman inspect "${IMAGE_NAME}" | grep kernel-core
+        "created_by": "kernel-core-5.14.0-427.26.1.el9_4.x86_64"
+```
+
+When a `bootc` image is started, it uses the host kernel, which is not necessarily
+the same one used for building the image. This means that the `openvswitch` module
+cannot be loaded in the container due to the kernel version mismatch with the
+modules present in the `/lib/modules` directory.
+
+To work around this problem, it is necessary to share the host `/lib/modules`
+directory with the container when it is started. This is accomplished by providing
+the `-v /lib/modules:/lib/modules:ro` option to the `podman run` command as
+described below.
+
+### Configure CSI
+
+If the host is already configured to have a `rhel` volume group with free space,
+this configuration is inherited by the container so that it can be used by the
+MicroShift CSI driver to allocate storage.
+
+```
+$ sudo vgs
+  VG   #PV #LV #SN Attr   VSize   VFree
+  rhel   1   1   0 wz--n- <91.02g <2.02g
+```
+
+Additional configuration is required for using storage in `bootc` MicroShift
+images when the host does not have the volume group setup as required by the
+MicroShift CSI driver.
+
+Run the following commands to create a file to be used for LVM partitioning and
+set it up as a loop device.
+
+```
+VGFILE=/var/lib/microshift-lvm-storage.img
+VGSIZE=1G
+
+sudo truncate --size="${VGSIZE}" "${VGFILE}"
+sudo losetup -f "${VGFILE}"
+```
+
+Query the loop device name and create a free volume group on the device according
+to the MicroShift CSI driver requirements.
+
+```
+VGLOOP=$(losetup -j ${VGFILE} | cut -d: -f1)
+sudo vgcreate -f -y rhel "${VGLOOP}"
+```
+
+The device will now be shared with the newly created containers as described in
+the next section.
+
+> The following commands can be run to detach the loop device and delete the LVM
+> volume group file.
+> ```
+> sudo losetup -d "${VGLOOP}"
+> sudo rm -f "${VGFILE}"
+> ```
+
+### Run Container
+
+Run the following command to start the MicroShift `bootc` image in an interactive
+terminal session. Note that the host container storage and kernel modules paths
+are shared with the container.
+
+```
+IMAGE_NAME=microshift-4.16-bootc
+
+sudo podman run --rm -it --privileged \
+    -v /var/lib/containers/storage:/var/lib/containers/storage \
+    -v /lib/modules:/lib/modules:ro \
+    --name "${IMAGE_NAME}" \
+    "${IMAGE_NAME}"
+```
+
+> If additional LVM volume group device was allocated as described in the
+> [Configure CSI](#configure-csi) section, the loop device should automatically
+> be shared with the container and used by the MicroShift CSI driver.
+
+Log into the container host when prompted using the `redhat:redhat` credentials.
```

Comment:
"container host" isn't defined previously and is somewhat confusing. Maybe this can be reworded such as:

> After the MicroShift `bootc` image has been successfully started, there will be a login prompt presented. Login to this running container using the `redhat:redhat` credentials.